src/index/hnsw/faiss_hnsw.cc

```diff
@@ -827,15 +827,26 @@ class BaseFaissRegularIndexHNSWNode : public BaseFaissRegularIndexNode {
 };
 
 //
+template <typename DataType>
```

Comment:
No, definitely not!
Such a `template<>` leads to way larger code size, bcz corresponding identical class functions will be generated for each of the templated versions. Please leave it as it was written, because it was written with a purpose.